presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ExtractSpatialJoins.java

```diff
@@ -259,28 +254,43 @@ public boolean isEnabled(Session session)
         public Result apply(JoinNode joinNode, Captures captures, Context context)
         {
             Expression filter = joinNode.getFilter().get();
-            List<FunctionCall> spatialFunctions = extractSupportedSpatialFunctions(filter);
+            return tryCreateSpatialJoin(context, joinNode, filter, joinNode.getId(), joinNode.getOutputSymbols(), metadata, splitManager, pageSourceManager, sqlParser);
```

Comment:
I think this refactoring is not needed. (If it was needed, it would need to be in a separate commit.)